common/utilities/string/string-utilities.h

```diff
@@ -0,0 +1,103 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+
+namespace utilities {
+    namespace string {
+
+        // Converts string to number
+        // returns true if conversion succeeded, false otherwise
+        // Input: 
+        //      - string of a number
+        //      - parameter T - the number's type
+
+        template<class T>
+        inline bool string_to_value(const std::string& str, T& result)
+        {
+            try
+            {
+                size_t last_char_idx;
+                if (std::is_arithmetic<T>::value)
```

Comment:
Please adjust the function that `std::is_arithmetic` will be checked during compile and  not runtime with 
`template <typename T, typename std::enable_if<std::is_arithmetic<T>`
https://stackoverflow.com/questions/44848011/c-limit-template-type-to-numbers